### ISSUE TITLE
Feature Request : Official Parcel Support

### ISSUE THREAD

**Issue by tatsujb** [angular/angular#26822]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

I saw that dev is taking off into a route that I really like what with angular 6 then 7 and soon 8 (possibly bearing official IVy support, on which I'm particularly keen), and I can't help but wonder, where's the parcel love?

parcel should work with angular and was build with the goal of doing so. 

But the way things have turned out, a year after the fact, parcel is still a react/preact/vue and typsecript bundler but not an angular one.

I feel as though the parcel team are 100% compliant and willing to comply further but just don't know what exactly the angular team want from them. And, to me, it seems the parcel team have been in the dark.

is it a will of angular to disallow switching to parcel?

Am I just understanding something wrong and parcel would just not fit in the angular-cli stack because of some technical reason?

I've tried switching my project over to parcel ceaselessly for a year now and nothing can be done it seems entirely incompatible.

it seems to me much like the official rxjs and webpack support angular is providing it would be fair to turn dev attention to parcel.

(btw I'm not suggesting angular-cli ng commands are to be run by parcel, obviously not. to me that's not where parcel's intrests lies, so I'm perfectly fine with having a stack that utilizes both webpack (in the ng commands) and parcel (for compile and serve) where each has a strong suit. As a result no dev time needs to be added for angular-cli)

I don't feel like it would be very difficult to make angular and parcel compatible.

thanks for reading :)

**Comment by sarunint** [comment]
@tatsujb I'm not sure that Parcel support compilation pipeline? For production builds, Angular needs to compile it's template down to plain JavaScript (using `@angular/compiler`. So all decorators is gone from the sources. Could you elaborate more on that point?

**Comment by trotyl** [comment]
@tatsujb You should possibly ask Parcel to add support for Angular, [all of the official "plugins" are maintained by Parcel itself](https://github.com/parcel-bundler/parcel/tree/master/packages/core/parcel-bundler/src/assets). Webpack has an open plugin system, so that plugin can he made by Angular side.

Technically any tool supporting TypeScript could support Angular compiler, they're using the similar CLI interface and JavaScript API.

**Comment by tatsujb** [comment]
yeah. I also just opened an issue on their side.

I know that it's more their job, but I was wondering if we knocked heads if we could come up with a reason for why it doesn't exactly work, because, yes, on paper, it 100% should.

**Comment by robwormald** [comment]
I've looked into this, and spoken briefly with the Parcel folks about it. 

tldr: this is certainly doable, but its complicated. The Ivy compiler is a significantly simpler model and easier to integrate with something like Parcel, as it works a lot more like the regular TypeScript compiler. 

This should really be opened up on the https://github.com/angular/angular-cli repo, as its tooling related. I don't think we're going to officially support Parcel in the short term, but happy to support this as a community effort.
